lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -206,43 +218,57 @@ withNetworkLayer
         -- ^ Callback function with the network layer
     -> IO a
 withNetworkLayer tr np addrInfo versionData action = do
-    localTxSubmissionQ <- atomically newTQueue
-
-    -- NOTE: We keep a client connection running for accessing the node tip,
-    -- submitting transactions, and querying parameters.
+    -- NOTE: We keep client connections running for accessing the node tip,
+    -- submitting transactions, querying parameters and delegations/rewards.
+    --
     -- It is safe to retry when the connection is lost here because this client
-    -- doesn't really do anything but sending dummy messages to get the node's
-    -- tip. It doesn't rely on the intersection to be up-to-date.
-    nodeTipVar <- atomically $ newTVar TipGenesis
-    protocolParamsVar <- atomically $ newTVar $ W.protocolParameters np
-    nodeTipClient <- mkTipSyncClient tr np
-        localTxSubmissionQ
-        (atomically . writeTVar nodeTipVar)
-        (atomically . writeTVar protocolParamsVar)
+    -- doesn't really do anything but sending messages to get the node's tip. It
+    -- doesn't rely on the intersection to be up-to-date.
     let handlers = retryOnConnectionLost tr
-    link =<< async
-        (connectClient tr handlers nodeTipClient versionData addrInfo)
+
+    (nodeTipVar, protocolParamsVar, localTxSubmissionQ) <-
+        connectNodeTipClient handlers
+
+    queryRewardQ <- connectDelegationRewardsClient handlers
 
     action
         NetworkLayer
             { currentNodeTip = liftIO $ _currentNodeTip nodeTipVar
             , nextBlocks = _nextBlocks
-            , initCursor = _initCursor
+            , initCursor = _initCursor queryRewardQ
```

Comment:
Wait, why is `queryRewardQ` both read from by the shared `connectDelegationRewardsClient` and each wallet client? That seems odd. Shouldn't `queryRewardQ` _only_ be handled by the shared client?

Reply:
Oops. I meant to remove that.